File: sdks/javascript/examples/credential-verification.ts
/**
 * Example: Email verification and credential management.
 *
 * Run with: npx tsx examples/credential-verification.ts
 */

import { ATAPClient } from "@atap/sdk";

async function main() {
  // Create an authenticated client
  const client = new ATAPClient({
    baseUrl: "http://localhost:8080",
    did: "did:web:localhost%3A8080:human:user",
    privateKey: "<base64 Ed25519 seed>",
    clientSecret: "atap_...",
  });

  // Start email verification
  const msg = await client.credentials.startEmailVerification(
    "user@example.com",
  );
  console.log(`Email verification: ${msg}`);

  // Verify with OTP (user enters code from email)
  const credential = await client.credentials.verifyEmail(
    "user@example.com",
    "123456",
  );
  console.log("Email verified!");
  console.log(`  Credential ID:   ${credential.id}`);
  console.log(`  Type:            ${credential.type}`);
  console.log(`  Issued at:       ${credential.issuedAt}`);

  // List all credentials
  const credentials = await client.credentials.list();
  console.log(`\nTotal credentials: ${credentials.length}`);
  for (const cred of credentials) {
    console.log(`  - ${cred.type} (${cred.id})`);
  }

  // Check credential status list (public endpoint)
  const status = await client.credentials.statusList("1");
  console.log(`\nStatus list: ${JSON.stringify(status)}`);
}

main().catch(console.error);

File: sdks/python/examples/credential_verification.py
"""Example: Email verification and credential issuance."""

from atap import ATAPClient

# Create an authenticated client (human entity)
client = ATAPClient(
    base_url="http://localhost:8080",
    did="did:web:localhost%3A8080:human:abc123",
    private_key="<base64 private key>",
)
client.token_manager.obtain_authorization_code()

# Start email verification (sends OTP)
msg = client.credentials.start_email_verification("user@example.com")
print(msg)  # "OTP sent to email"

# In production: user receives OTP via email
# For local dev: check server logs for the OTP
otp = input("Enter OTP: ")

# Verify email and receive credential
credential = client.credentials.verify_email("user@example.com", otp)
print(f"Credential issued: {credential.type}")
print(f"VC JWT: {credential.credential[:50]}...")

# List all credentials
all_creds = client.credentials.list()
for c in all_creds:
    print(f"  - {c.type} (issued: {c.issued_at})")

client.close()
